kyaml/kio/kio.go

```diff
@@ -88,7 +90,13 @@ func (p Pipeline) Execute() error {
 		if err != nil {
 			return errors.Wrap(err)
 		}
-		result = append(result, nodes...)
+		for _, node := range nodes {
```

Comment:
nit: consider doing `for i := range nodes` and `append(result, nodes[I])`.

Reply:
How would this work if there are 2 nodes in a file and 1 with no process, other without? I doubt it yields expected behavior.

Reply:
Good point, we probably need to add these back in at the end.  This is similar to how scoping adds them back in.

As written, this will delete any no-process objects from the file.

Reply:
Suppressing reading also suppresses validation, formatting, collecting statistics, etc.

Reply:
Yes. We should not delete those objects in such case.

Reply:
> Good point, we probably need to add these back in at the end. This is similar to how scoping adds them back in.
> 
> As written, this will delete any no-process objects from the file.

Yeah--good point. I've re-implemented to store the no-process nodes and re-add them after the filter processing. Please have look and let me know what you think.

Reply:
> nit: consider doing `for i := range nodes` and `append(result, nodes[I])`.

Done